test/cartesian/XAxis.spec.tsx

```diff
@@ -373,190 +373,243 @@ describe('<XAxis />', () => {
-      const { container } = render(
-        <BarChart width={300} height={300} data={data}>
-          <XAxis dataKey="x" type="number" tickCount={3} />
-        </BarChart>,
-      );
-      expectXAxisTicks(container, ['0', '85', '170']);
-    });
+      it('should allow providing more tickCount', () => {
+        const { container } = render(
+          <Component>
+            <XAxis dataKey="x" type="number" tickCount={7} />
+          </Component>,
+        );
+        expectXAxisTicks(container, ['0', '30', '60', '90', '120', '150', '180']);
+      });
 
-    it('should make ticks from dataMin, dataMax', () => {
-      const { container } = render(
-        <BarChart width={300} height={300} data={data}>
-          <XAxis dataKey="x" type="number" domain={['dataMin', 'dataMax']} allowDataOverflow />
-        </BarChart>,
-      );
-      expectXAxisTicks(container, ['100', '120', '140', '170']);
-    });
+      it('should allow providing less tickCount', () => {
+        const { container } = render(
+          <Component>
+            <XAxis dataKey="x" type="number" tickCount={3} />
+          </Component>,
+        );
+        expectXAxisTicks(container, ['0', '85', '170']);
+      });
 
-    it('should default to dataMin, dataMax ANYWAY when domain is provided as strings? I do not understand this behaviour.', () => {
-      const { container } = render(
-        <BarChart width={300} height={300} data={data}>
-          {/* omg recharts just completely ignores this when it's defined as strings */}
-          <XAxis dataKey="x" type="number" domain={['-500', '500']} allowDataOverflow />
-        </BarChart>,
-      );
-      expectXAxisTicks(container, ['100', '120', '140', '170']);
-    });
+      it('should make ticks from dataMin, dataMax', () => {
+        const { container } = render(
+          <Component>
+            <XAxis dataKey="x" type="number" domain={['dataMin', 'dataMax']} allowDataOverflow />
+          </Component>,
+        );
+        expectXAxisTicks(container, ['100', '120', '140', '170']);
+      });
 
-    it('should allow a function that returns a domain, and pass inside a computed domain and allowDataOverflow prop', () => {
-      const spy = vi.fn();
-      spy.mockReturnValue([-500, 500]);
-      const { container, rerender } = render(
-        <BarChart width={300} height={300} data={data}>
-          <XAxis dataKey="x" type="number" domain={spy} allowDataOverflow />
-        </BarChart>,
-      );
-      expectXAxisTicks(container, ['-500', '-250', '0', '250', '500']);
-      expect(spy).toHaveBeenCalledTimes(1);
-      expect(spy).toHaveBeenCalledWith([100, 170], true);
+      it('should default to dataMin, dataMax ANYWAY when domain is provided as strings? I do not understand this behaviour.', () => {
+        const { container } = render(
+          <Component>
+            {/* omg recharts just completely ignores this when it's defined as strings */}
```

Comment:
I mean, the domain is supposed to be numbers and it isn't. Would you expect it to parse the strings as numbers or maybe throw an error?

Reply:
It parses strings elsewhere

Reply:
true that, I think D3 is what parses strings though? At least string data..

Reply:
Yep. Although I would always always steer users away from strings if I could either way